sei-protocol/sei-chain
# parallelization test contracts & fixes

## Describe your changes and provide context
Add in contracts used for testing parallelization.

As made some fixes to dependencies in bank & dex message types

## Testing performed to validate your change
tested by
1. deploying the contracts to local chain
2. register dependencies according to `deps.json` under each contract's directory
3. add in logging before and after `DeliverTx`
4. send multiple transactions in the same block (e.g. with `async` broadcast type) and observe the logs. If there are overlaps between logs for different transactions, we know that the transactions are running in parallel.
